test/e2e/storage/testsuites/provisioning.go

```diff
@@ -198,14 +208,34 @@ func testProvisioning(input *provisioningTestInput) {
 		input.pvc.Spec.VolumeMode = &block
 		TestDynamicProvisioning(input.testCase, input.cs, input.pvc, input.sc)
 	})
+
+	It("should provision storage with snapshot data source [Feature:VolumeSnapshotDataSource]", func() {
+		if !input.dInfo.Capabilities[CapDataSource] {
+			framework.Skipf("Driver %q does not support populate data from snapshot - skipping", input.dInfo.Name)
+		}
+		if input.dInfo.Name == "csi-hostpath-v0" {
+			framework.Skipf("skip test when using driver csi-hostpath-v0 - skipping")
+		}
+
+		input.testCase.SkipWriteReadCheck = true
```

Comment:
I think it would be valuable to do read/write validation with snapshots.  Actually, a good test case could be to:
1. Write some data
2. Take a snapshot
3. Write more data
4. Create a new volume from snapshot
5. Validate data is the old data

Reply:
Can we cover that by a new issue and merge this PR after addressing your other concern?

Reply:
sure that's fine